From 4d191b0af0c9fc3d5c3daf82e2751b97ee1b6bbc Mon Sep 17 00:00:00 2001
From: ningyu <ningyu@eswincomputing.com>
Date: Thu, 7 Aug 2025 21:20:20 +0800
Subject: [PATCH 469/533] feat:implement pin header function of D560

Changelogs:
Modify dts in order to implement pin header functions of D560

Signed-off-by: ningyu <ningyu@eswincomputing.com>
---
 arch/riscv/boot/dts/eswin/eic7702-d560.dts | 133 ++++++++++++++-------
 1 file changed, 92 insertions(+), 41 deletions(-)

diff --git a/arch/riscv/boot/dts/eswin/eic7702-d560.dts b/arch/riscv/boot/dts/eswin/eic7702-d560.dts
index 5f06d6743016..a58f5817d75d 100644
--- a/arch/riscv/boot/dts/eswin/eic7702-d560.dts
+++ b/arch/riscv/boot/dts/eswin/eic7702-d560.dts
@@ -85,17 +85,6 @@ &d1_sdio0 {
 	no-sd;
 	no-mmc;
 	vqmmc-supply = <&reg_vcc1v8>;
-	aw315:wifi_aw3155@0 {
-		compatible = "aml_w1_sdio";
-		reg = <0x0>;
-		interrupt-parent = <&d1_porta>;
-		interrupts = <18 IRQ_TYPE_LEVEL_LOW>;
-		pinctrl-names = "default", "default";
-		pinctrl-0 = <&pinctrl_die1_gpio18_default>;
-		pinctrl-1 = <&pinctrl_die1_gpio16_default>;
-		irq-gpios = <&d1_porta 18 GPIO_ACTIVE_HIGH>;
-		rst-gpios = <&d1_porta 16 GPIO_ACTIVE_HIGH>;
-	};
 	mdhd_wlan {
 		compatible = "android,bcmdhd_wlan";
 		pinctrl-names = "default";
@@ -105,9 +94,7 @@ mdhd_wlan {
 };
 
 &d0_usbdrd3_1 {
-	status = "okay";
-	pinctrl-names = "default";
-	pinctrl-0 = <&pinctrl_die0_gpio14_default>;
+	pinctrl-0 = <&pinctrl_die0_usb1_pwren_default>, <&pinctrl_die0_gpio14_default>;
 	hub-rst-gpios = <&porta 14 GPIO_ACTIVE_HIGH>;
 };
 
@@ -122,17 +109,35 @@ &d0_uart2 {
 };
 
 &d1_uart2 {
+	/* unused */
+	status = "disabled";
+};
+
+&d0_uart3 {
 	/* pin header */
 	status = "okay";
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_die0_uart3_default>;
 };
 
 &ssi0 {
 	/* pin header */
 	status = "okay";
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_die0_dual_spi1_default>;
+	num-cs = <1>;
+	spi-flash@0 {
+		compatible = "jedec,spi-nor";
+		reg = <0>;
+		#address-cells = <1>;
+		#size-cells = <1>;
+		spi-max-frequency = <4800000>;
+		rx-sample-delay-ns = <10>;
+	};
 };
 &d1_ssi0 {
-	/* pin header */
-	status = "okay";
+	/* unused */
+	status = "disabled";
 };
 
 &d0_graphcard1 {
@@ -179,8 +184,6 @@ &fan_control {
 };
 
 &d1_fan_control {
-	pinctrl-0 = <&pinctrl_die1_fan_tach_default &pinctrl_die1_pwm0_default
-				&pinctrl_die1_pwm1_default &pinctrl_die1_pwm2_default>;
 	status = "okay";
 	fan-0 {
 		label = "d1_fan";
@@ -200,8 +203,8 @@ fan-2 {
 };
 
 &d0_i2c1 {
-	/* pin header */
-	status = "okay";
+	/* unused */
+	status = "disabled";
 };
 &d1_i2c1 {
 	/* eeprom(at24c02c) & rtc(rs4c411) */
@@ -237,13 +240,23 @@ &d1_i2c3 {
 
 &d0_i2c4 {
 	/* d0 csi2-3 */
+	status = "okay";
 };
+
 &d1_i2c4 {
 	/* d1 csi2-3 */
+	status = "okay";
 };
 
 &d0_i2c5 {
 	/* pin header */
+	status = "okay";
+	tmp102@48 {
+		compatible = "ti,tmp102";
+		reg = <0x48>;
+		label = "board_tmp";
+		#thermal-sensor-cells = <1>;
+	};
 };
 &d1_i2c5 {
 	/* cc logic fusb303b */
@@ -266,17 +279,28 @@ connector {
 };
 
 &d0_i2c8 {
-	/* pin header */
+	/* unused */
+	status = "disabled";
 };
 &d1_i2c8 {
-	/* pin header */
+	/* unused */
+	status = "disabled";
 };
 
 &d0_i2c9 {
-	/* pin header */
+	/* unused */
+	status = "disabled";
 };
 &d1_i2c9 {
 	/* pin header */
+	status = "okay";
+	sys_power:ina226@40 {
+		compatible = "ti,ina226";
+		#io-channel-cells = <1>;
+		label = "sys_power";
+		reg = <0x40>;
+		shunt-resistor = <1000>;
+	};
 };
 
 &wdt0 {
@@ -330,31 +354,58 @@ &d1_pcie {
 };
 
 &pinctrl {
+	/* pin header */
 	status = "okay";
 	pinctrl-names = "default";
-	pinctrl-0 = <&pinctrl_die0_gpio5_default &pinctrl_die0_gpio6_default &pinctrl_die0_gpio12_default &pinctrl_die0_gpio35_default
-				 &pinctrl_die0_gpio39_default &pinctrl_die0_gpio40_default &pinctrl_die0_gpio41_default &pinctrl_die0_gpio46_default
-				 &pinctrl_die0_gpio47_default &pinctrl_die0_gpio54_default &pinctrl_die0_gpio55_default &pinctrl_die0_gpio73_default
-				 &pinctrl_die0_gpio75_default &pinctrl_die0_gpio79_default &pinctrl_die0_gpio81_default &pinctrl_die0_gpio36_default
-				 &pinctrl_die0_gpio37_default &pinctrl_die0_gpio38_default &pinctrl_die0_gpio74_default &pinctrl_die0_gpio92_default
-				 &pinctrl_die0_gpio93_default>;
-
-	/* pin header default function for GPIO
-	   D0_SPI1 (CS0,SCLK,MOSI,MISO,D2,D3): GPIO 35,36,37,38,39,40
-	   D0_I2C1 (SCL,SDA): GPIO 46,47
-	   D0_I2C5 (SCL,SDA): GPIO 54,55
-	   D1_SPI1 (CS0,SCLK,D0,D1,D2): GPIO 35,36,37,38,39
-	   D1_UART3(TX,RX): GPIO 92,93
-	   D0_UART3(TX,RX): GPIO 92,93
-	*/
+	pinctrl-0 = <&pinctrl_die0_gpio6_default &pinctrl_die0_gpio12_default &pinctrl_die0_gpio39_default 
+				&pinctrl_die0_gpio40_default &pinctrl_die0_gpio41_default &pinctrl_die0_gpio46_default
+				&pinctrl_die0_gpio47_default &pinctrl_die0_gpio73_default &pinctrl_die0_gpio74_default
+				&pinctrl_die0_gpio75_default &pinctrl_die0_gpio81_default>;
+
 };
 
 &d1_pinctrl {
+	/* pin header */
 	status = "okay";
 	pinctrl-names = "default";
-	pinctrl-0 = <&pinctrl_die1_gpio35_default &pinctrl_die1_gpio36_default &pinctrl_die1_gpio37_default &pinctrl_die1_gpio38_default
-				 &pinctrl_die1_gpio39_default &pinctrl_die1_gpio92_default &pinctrl_die1_gpio93_default>;
-};
+	pinctrl-0 = <&pinctrl_die1_gpio36_default &pinctrl_die1_gpio39_default
+				&pinctrl_die1_gpio92_default &pinctrl_die1_gpio93_default
+				&pinctrl_die1_gpio18_default>;
+};
+
+/******** pin headers *********
+     pin : function
+       3 : D0 I2C5 SDA
+       5 : D0 I2C5 SCL
+       7 : D0 GPIO46
+       8 : D0 UART3 TX
+      10 : D0 UART3 RX
+      11 : D0 GPIO6
+      12 : D0 GPIO12
+      13 : D0 GPIO5 (x)
+      15 : D0 GPIO81
+      16 : D0 GPIO47
+      18 : D0 SPI1 D3(GPIO40)
+      19 : D0 SPI1 D0
+      21 : D0 SPI1 D1
+      22 : D0 SPI1 D2(GPIO39)
+      23 : D0 SPI1 CLK
+      24 : D0 SPI1 CS0
+      26 : D0 SPI1 CS1(GPIO41)
+      27 : D1 I2C9 SDA
+      28 : D1 I2C9 SCL
+      29 : D1 GPIO93
+      31 : D1 GPIO92
+      32 : D1 GPIO35 (x)
+      33 : D1 GPIO36
+      35 : D1 GPIO39
+      36 : D0 GPIO79 (x)
+      37 : D0 GPIO74
+      38 : D0 GPIO75
+      40 : D0 GPIO73
+*********************************/
+
+
 
 /* die0 GPIO Function Description
 
-- 
2.52.0

